aptos-move/aptos-vm/src/transaction_metadata.rs

```diff
@@ -34,6 +35,7 @@ pub struct TransactionMetadata {
     pub is_keyless: bool,
     pub entry_function_payload: Option<EntryFunction>,
     pub multisig_payload: Option<Multisig>,
+    pub automation_payload: Option<AutomationTransactionPayload>,
```

Comment:
if we do not see a need to have access to `AutomationTransactionPayload` from move-context, then this can be simply substituted with boolean flag indicating the origin of the entry-function call.

Reply:
@aregng , I was wondering if instead of a `bool`, we can have an `enum`  that indicates the originating tx type. That way tomorrow if it is a tx of a different kind and we need to identify from `SafeNativeContext`, we can do so without adding one more field here.